sdcm/provision/azure/utils.py

```diff
@@ -25,6 +26,14 @@
 LOGGER = logging.getLogger(__name__)
 
 
+def date_or_int_str_to_int(build_id) -> int:
+    """Converts int string or date string to int (timestamp for date) for comparison."""
+    try:
+        return int(build_id)
+    except ValueError:
+        return int(dateutil.parser.parse(build_id, ignoretz=True).timestamp())
```

Comment:
I don't think the date in the build-id was intended, and it's an artifact of some test builds that in other backend we filter by tags

i.e. we should fix it at the core.

I think it would be safe enough to ignore those , or just give them zero, don't assume they would be date

Reply:
anyway it is done - no need to downgrade to more primitive solution. If the problem is at the core, lets fix there, but anyway images list is spoiled and we need to keep it.

Reply:
I'll stress that again
"don't assume they would be date",
if the contract you think it should is broken, ignore it.
you just create a new assumption that it might be a date, tomorrow someone would put there just "testing" and again this image retrieval code would be broken

Reply:
anytime someone would build manually and not in jenkins job, it might have something else, not only a date:

https://github.com/scylladb/scylla-machine-image/blob/34636d1c8402df76c2d8de4df0ec2f732a425b0d/packer/build_deb_image.sh#L38

Reply:
BTW, one more reason to drop those without a proper build-id, means that we never want to select those manually created images (happened for use before in AWS, and we were running the weekly on images someone built locally, and not the official latest build, took us a while to figure that out)

Reply:
meanwhile, I'm merging, since it's blocking me from running provision test on PRs.
but I still think we should filter the one without `build-id`